homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -113,6 +140,35 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
+async def async_migrate_entries(
+    hass: HomeAssistant,
+    config_entry_id: str,
+    entry_callback: Callable[[RegistryEntry], dict[str, Any] | None],
+) -> None:
+    """Migrator of unique IDs.
+
+    Copied and adjusted from `homeassistant.helpers.entity_registry.async_migrate_entries`
+    as migration will fail (and therefore component will not load) if the new unique_id
+    has already been created (i.e. using a custom component or dev version).
```

Comment:
I don't think we should copy this whole function here. Just add a check for an existing entry with the new unique and return None in that case in the callback.

Reply:
Oh, does the callback has access to the registry to check if the id exists? As far as I can tell neither a `hass` object nor an `EntityRegistry` is available there?

Reply:
Since we define the callback within `async_setup_entry` it has access to `hass` and any other name in the outer scope.

Reply:
Ah, thanks! Learned something today :) I'll implement it accordingly.